### ISSUE TITLE
various error during pip install and import 

### ISSUE THREAD

**Issue by sugizo** [EchterAlsFake/PHUB#55]
**env** 
google colab

# trial 1
**step**
`pip install --upgrade git+https://github.com/EchterAlsFake/PHUB.git@py-3.10`

**code**
`import phub`

**result**
```
AttributeError                            Traceback (most recent call last)
[<ipython-input-5-09043198e867>](https://localhost:8080/#) in <cell line: 1>()
----> 1 import phub

2 frames
[/usr/local/lib/python3.10/dist-packages/phub/utils.py](https://localhost:8080/#) in <module>
    156             for key in keys}
    157 
--> 158 def suppress(gen: Iterable, errs: Exception | tuple[Exception] = errors.VideoError) -> Iterator:
    159     '''
    160     Set up a generator to bypass items that throw errors.

AttributeError: module 'phub.errors' has no attribute 'VideoError'
```

# trial 2
**execute**
`pip install --upgrade git+https://github.com/EchterAlsFake/PHUB.git`

**result**
```
Collecting git+https://github.com/EchterAlsFake/PHUB.git
  Cloning https://github.com/EchterAlsFake/PHUB.git to /tmp/pip-req-build-sbfyqxqd
  Running command git clone --filter=blob:none --quiet https://github.com/EchterAlsFake/PHUB.git /tmp/pip-req-build-sbfyqxqd
  Resolved https://github.com/EchterAlsFake/PHUB.git to commit 7791aee995424e6a42206bd889659f99d007b566
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: requests in /usr/local/lib/python3.10/dist-packages (from phub==4.5.7) (2.31.0)
Collecting ffmpeg-progress-yield (from phub==4.5.7)
  Downloading ffmpeg_progress_yield-0.7.8-py2.py3-none-any.whl (10 kB)
INFO: pip is looking at multiple versions of phub to determine which version is compatible with other requirements. This could take a while.
ERROR: Package 'phub' requires a different Python: 3.10.12 not in '>=3.11'
```

# trial 3
`pip install --upgrade phub`

**result**
still got the previous reported error https://github.com/EchterAlsFake/PHUB/issues/54

best regards

**Comment by EchterAlsFake** [comment]
Okay seems like the problems are a bit deeper. I'm gonna work on it in 4 hours.

**Comment by Egsagon** [comment]
**Trial 1:**
I cannot replicate the error. py-3.10 branch works fine for me; is it due to your env or interpreter (even tho i don't think that's the issue)?

**Trial 2:**
The master branch only supports python 3.11+, but you are trying to install it from python 3.10. Either change your python version, use the py-3.10 branch (which i am dropping support for) or the compat branch available from python 3.9+.

**Trial 3:**
Replicates #54 because @EchterAlsFake didn't opened a release after making the changes. Use the trial 2 URL instead.

**Comment by sugizo** [comment]
like written in first posted the environment is `google colab`
the detail are : 
```
os : ubuntu jammy
python : 3.10.12
```

that's why didnt do the trial for `compat` that is for python lower version

**Comment by Egsagon** [comment]
The py-3.10 branch doesn't have a reason to live anymore and will probably disapear soon, will be replaced by the compat branch. Lower python version does not mean it's not compatible with the most recent ones.

**Comment by EchterAlsFake** [comment]
So I just also tested it in my environment and it works fine for me. Sometimes the pip --upgrade is a bit broken. Please completely uninstall phub and reinstall the 3.10 branch:
```
pip uninstall phub
pip install git+https://github.com/EchterAlsFake/PHUB@py-3.10
```

Let me know if it worked or not :)

**Comment by sugizo** [comment]
`pip install git+https://github.com/EchterAlsFake/PHUB.git@py-3.10`
work

but
`pip install phub`
still got the previous reported error https://github.com/EchterAlsFake/PHUB/issues/54

best regards

**Comment by Egsagon** [comment]
As said, the fix for #54 has not been released yet. If you want to use the 3.11 branch, use `pip install git+https://github.com/EchterAlsFake/phub`. Hope this solves your problem!
